pkg/api/vm/handler.go

```diff
@@ -227,6 +239,19 @@ func (h *vmActionHandler) doAction(rw http.ResponseWriter, r *http.Request) erro
 		}
 	case dismissInsufficientResourceQuota:
 		return h.dismissInsufficientResourceQuota(name, namespace)
+	case updateResourceQuotaAction:
+		if ok, err := apiutil.CanUpdateResourceQuota(h.clientSet, namespace, user.GetName()); err != nil {
+			return apierror.NewAPIError(validation.ServerError, fmt.Sprintf("Failed to check permission: %v", err))
+		} else if !ok {
+			return apierror.NewAPIError(validation.PermissionDenied, "User does not have permission to update resource quota")
+		}
```

Comment:
Since `CanUpdateResourceQuota()` is already checked in https://github.com/FrankYang0529/harvester/blob/5b5235db6156776a16f18044c4a5015022bc5200/pkg/api/vm/formatter.go#L57-L65, is it necessary to check again here?

Reply:
Yeah, I see. I think `Formatter` is just to show which actions can be applied to this VM, not a real authentication. If an user has permission to operate VM, they still can do the operation even the action is not showed in the `Formatter`.

The steve server do VM permission check when users call a VM URL. However, `ResourceQuota` is a different object, so default steve server checks of VM permission is not enough. That's why we still have to do a check again here.